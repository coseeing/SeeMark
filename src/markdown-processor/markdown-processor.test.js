import '@testing-library/jest-dom';
import { getByRole } from '@testing-library/dom';

import createDOMFromHTML from '../testing-helpers/create-dom-from-html';

import markdownProcessor from './';

describe('markdownProcessor', () => {
  it('should process markdown content as desired', () => {
    const markdownContent = '# Hello World';
    const options = {
      latexDelimiter: 'bracket',
      documentFormat: 'inline',
      imageFiles: {},
    };

    const result = markdownProcessor(markdownContent, options);

    const container = createDOMFromHTML(result);

    const heading = getByRole(container, 'heading', {
      level: 1,
      name: 'Hello World',
    });

    expect(heading).toBeTruthy();
  });
});
