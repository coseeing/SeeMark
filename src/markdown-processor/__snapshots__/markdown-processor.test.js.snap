// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`markdownProcessor should handle math expressions with brackets 1`] = `
<div>
  <p>
    <span
      class="sr-only"
    >
      <math
        xmlns="http://www.w3.org/1998/Math/MathML"
      >
        
  
        <mrow
          data-mjx-texclass="ORD"
        >
          
    
          <msup>
            
      
            <mrow
              data-mjx-texclass="ORD"
            >
              
        
              <mrow
                data-mjx-texclass="INNER"
              >
                
          
                <mo
                  data-mjx-texclass="OPEN"
                >
                  (
                </mo>
                
          
                <mo>
                  âˆ’
                </mo>
                
          
                <mn>
                  3
                </mn>
                
          
                <mo
                  data-mjx-texclass="CLOSE"
                >
                  )
                </mo>
                
        
              </mrow>
              
      
            </mrow>
            
      
            <mrow
              data-mjx-texclass="ORD"
            >
              
        
              <mn>
                3
              </mn>
              
      
            </mrow>
            
    
          </msup>
          
  
        </mrow>
        

      </math>
    </span>
    <span
      aria-hidden="true"
    >
      <mjx-container
        class="MathJax"
        jax="SVG"
      >
        <svg
          focusable="false"
          height="2.709ex"
          role="img"
          style="vertical-align: -0.566ex;"
          viewBox="0 -947.3 2492.6 1197.3"
          width="5.639ex"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <path
              d="M94 250Q94 319 104 381T127 488T164 576T202 643T244 695T277 729T302 750H315H319Q333 750 333 741Q333 738 316 720T275 667T226 581T184 443T167 250T184 58T225 -81T274 -167T316 -220T333 -241Q333 -250 318 -250H315H302L274 -226Q180 -141 137 -14T94 250Z"
              id="MJX-1-TEX-N-28"
            />
            <path
              d="M84 237T84 250T98 270H679Q694 262 694 250T679 230H98Q84 237 84 250Z"
              id="MJX-1-TEX-N-2212"
            />
            <path
              d="M127 463Q100 463 85 480T69 524Q69 579 117 622T233 665Q268 665 277 664Q351 652 390 611T430 522Q430 470 396 421T302 350L299 348Q299 347 308 345T337 336T375 315Q457 262 457 175Q457 96 395 37T238 -22Q158 -22 100 21T42 130Q42 158 60 175T105 193Q133 193 151 175T169 130Q169 119 166 110T159 94T148 82T136 74T126 70T118 67L114 66Q165 21 238 21Q293 21 321 74Q338 107 338 175V195Q338 290 274 322Q259 328 213 329L171 330L168 332Q166 335 166 348Q166 366 174 366Q202 366 232 371Q266 376 294 413T322 525V533Q322 590 287 612Q265 626 240 626Q208 626 181 615T143 592T132 580H135Q138 579 143 578T153 573T165 566T175 555T183 540T186 520Q186 498 172 481T127 463Z"
              id="MJX-1-TEX-N-33"
            />
            <path
              d="M60 749L64 750Q69 750 74 750H86L114 726Q208 641 251 514T294 250Q294 182 284 119T261 12T224 -76T186 -143T145 -194T113 -227T90 -246Q87 -249 86 -250H74Q66 -250 63 -250T58 -247T55 -238Q56 -237 66 -225Q221 -64 221 250T66 725Q56 737 55 738Q55 746 60 749Z"
              id="MJX-1-TEX-N-29"
            />
          </defs>
          <g
            fill="currentColor"
            stroke="currentColor"
            stroke-width="0"
            transform="scale(1,-1)"
          >
            <g
              data-mml-node="math"
            >
              <g
                data-mjx-texclass="ORD"
                data-mml-node="TeXAtom"
              >
                <g
                  data-mml-node="msup"
                >
                  <g
                    data-mjx-texclass="ORD"
                    data-mml-node="TeXAtom"
                  >
                    <g
                      data-mjx-texclass="INNER"
                      data-mml-node="TeXAtom"
                    >
                      <g
                        data-mml-node="mo"
                      >
                        <use
                          data-c="28"
                          xlink:href="#MJX-1-TEX-N-28"
                        />
                      </g>
                      <g
                        data-mml-node="mo"
                        transform="translate(389,0)"
                      >
                        <use
                          data-c="2212"
                          xlink:href="#MJX-1-TEX-N-2212"
                        />
                      </g>
                      <g
                        data-mml-node="mn"
                        transform="translate(1167,0)"
                      >
                        <use
                          data-c="33"
                          xlink:href="#MJX-1-TEX-N-33"
                        />
                      </g>
                      <g
                        data-mml-node="mo"
                        transform="translate(1667,0)"
                      >
                        <use
                          data-c="29"
                          xlink:href="#MJX-1-TEX-N-29"
                        />
                      </g>
                    </g>
                  </g>
                  <g
                    data-mjx-texclass="ORD"
                    data-mml-node="TeXAtom"
                    transform="translate(2089,477.1) scale(0.707)"
                  >
                    <g
                      data-mml-node="mn"
                    >
                      <use
                        data-c="33"
                        xlink:href="#MJX-1-TEX-N-33"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </svg>
      </mjx-container>
    </span>
  </p>
  

</div>
`;
